File: isBalanced.java
public class isBalanced {
    public class TreeNode {
        int val;
        TreeNode left;
        TreeNode right;
        TreeNode() {}
        TreeNode(int val) { this.val = val; }
        TreeNode(int val, TreeNode left, TreeNode right) {
            this.val = val;
            this.left = left;
            this.right = right;
        }
    }
    
    public boolean isBalanced(TreeNode root) {
        if (root == null) {
            return true;
        } else if ((Math.abs(calculateHeight(root.left) - calculateHeight(root.right)) <= 1) && isBalanced(root.left) && isBalanced(root.right)) {
            return true;
        } else {
            return false;
        }
    }

    public int calculateHeight(TreeNode root) {
        if (root == null) {
            return 0;
        } else {
            int leftHeight = calculateHeight(root.left);
            int rightHeight = calculateHeight(root.right);
            return 1 + Math.max(leftHeight, rightHeight);
        }
    }
}

File: binaryTree/isBalanced.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def isBalanced(self, root: Optional[TreeNode]) -> bool:
        return (self.Height(root) >= 0)
        
    def Height(self, root):
        if root is None:
            return 0
        else:
            leftHeight, rightHeight = self.Height(root.left), self.Height(root.right) 
            if leftHeight < 0 or rightHeight < 0 or abs(leftHeight - rightHeight) > 1:
                return -1
            else:
                return max(leftHeight, rightHeight) + 1